### PR TITLE
:art: improve gitops template

### PR DESCRIPTION
**Pull request by flaxel**
### Type of Change

- [ ] Bugfix
- [x] Enhancement / new feature
- [ ] Refactoring
- [ ] Documentation

### Description

- add optional npm token to pull private npm packages
- rename docker image name

### Checklist

- [x] Add relevant labels (for example type of change or patch/minor/major)
- [x] Make sure not to introduce some mistakes
- [x] Update documentation
- [x] Review the [Contributing Guideline](../blob/main/CONTRIBUTING.md) and sign CLA
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by timkante** [comment]
I'm not a fan of this, as we put backend specific implementation detail into the shared workflows. Feels like an antipattern to me.

**Comment by flaxel** [comment]
> I'm not a fan of this, as we put backend specific implementation detail into the shared workflows. Feels like an antipattern to me.

@timkante Yeah okay, I didn't think of it that way. I think that any other repository that wants to use a gradle cache can also use these secrets. This makes it not repository specific for me and a possible solution to use the reusable workflow there too. There is also already a [discussion](https://github.com/github-community/community/discussions/17554) on the topic, where exactly the lack of the feature is addressed.

**Comment by timkante** [comment]
I think inlining the gradle-specifics into the docker-build-args on the call-side of the workflows is the better solution. I don't see the benefit of doing this "expansion" for a user in this workflow logic directly.

**Comment by flaxel** [comment]
@timkante Ah okay, now I know where the problem lies. That would be the nicer solution, but unfortunately that is not possible at the moment, because due secrets can only be used in the secrets block. So the current implementation is an interim solution and I hope from the mentioned discussion the feature will be provided for the reusable worflows as well as for the composite actions.

**Comment by timkante** [comment]
Merging this feels really bad, as this is mixing responsibilities. The caller should worry about how and with what the image is build. Adding this workaround for NPM and Gradle, moves logic of how images are built into this workflow, which I dislike. 

I won't approve this, I'm sorry. I'd rather wait for that feature then.

**Comment by flaxel** [comment]
@timkante That's okay, it's your opinion. 👍 It's not the prettiest solution, but I wanted to give other repos the ability to use the reusable workflows as well and have seen the implementation more often in some cases. Especially because it is uncertain when the feature will be implemented. This can take a very long time. If you want you can upvote the discussion to get more attention for the topic. 

My opinion is that the logic still lies with the caller, because the secrets must be specified. Without that, the configuration cannot be used either.

I would also be interested in the opinions of the other @Staffbase/workflow-enthusiasts.

**Comment by MarvinZeising** [comment]
Hm, this one's tricky, I think.
Regarding responsibilities, I think there's a difference between the npm token and the gradle caching stuff.

The npm token is only there to consume other GitHub repositories. So it's not bound to a certain technical implementation, just GitHub. So, having a GitHub action being able to read GitHub repositories makes sense to me.

However, the gradle cache is more of a technical implementation detail for me. A general GitHub deployment action probably shouldn't care too much about the implementation of what it's supposed to build and deploy. Especially, since the vast majority of our repos don't use grade, AFAIK.

I think we really need the possibility for a repository to pass any secrets (like the gradle stuff) to its own build step.

File: .github/workflows/template_gitops.yml [hunk 1]
```diff
@@ -58,9 +60,12 @@ jobs:
           dockerusername: ${{ secrets.docker_username }}
           dockerpassword: ${{ secrets.docker_password }}
           dockerenabled: ${{ inputs.dockerenabled }}
-          dockerbuildargs: ${{ inputs.dockerbuildargs }}
+          dockerbuildargs: |
+            ${{ inputs.dockerbuildargs }}
+            # remove token if feature is available: https://github.com/github-community/community/discussions/17554
```
Comment (timkante) [inline .github/workflows/template_gitops.yml:65]: Are you sure that this will work and the comment will not be passed as a build-arg to docker?

**Pull request merged**